### DISCUSSION TITLE
Error adding policy: failed to create pg.policies: operator does not exist: uuid = text

### DISCUSSION THREAD

**Discussion by lumenwrites** [supabase/supabase#7697] [Questions]
Hi! I have a `Profile` table that contains all the data I want to store about the logged in user. Every time a user signs up to my website using Supabase Auth, Postgres trigger function creates a new row in the `Profile` table, with an `id` matching the `id` of the new user. Here's what profile table looks like:
<img width="1118" alt="Screen Shot 2022-07-15 at 02 32 37" src="https://user-images.githubusercontent.com/4761235/179128882-b126f3ee-ab6c-484a-97cf-7541b8b94229.png">

Now my goal is to create a RLS policy that allows logged in user to fetch their profile data. Here's what I'm trying to do:
<img width="1131" alt="Screen Shot 2022-07-15 at 02 29 58" src="https://user-images.githubusercontent.com/4761235/179129014-8b566d2d-dd2f-4d80-8151-21602765aca5.png">
<img width="1145" alt="Screen Shot 2022-07-15 at 02 30 15" src="https://user-images.githubusercontent.com/4761235/179129020-0e9a409c-1b48-454c-bbb4-eca7be88aa7b.png">

But when I'm trying to save this policy, I'm getting an error:
<img width="352" alt="Screen Shot 2022-07-15 at 02 34 52" src="https://user-images.githubusercontent.com/4761235/179129101-f494498c-e53d-4418-9b05-b9f0bdeeb68b.png">

I'm new to RLS and I don't understand what am I doing wrong. Can you please help me to figure out how to do this correctly?

Maybe there's some kind of mismatch of types between `Profile.id` and `auth.users.id`? In that case, how can I make them match? The `Profile` table is created using Prisma, like so:
```
model Profile {
  id        String   @id @default(cuid())
  createdAt DateTime @default(now())
  updatedAt DateTime @default(now()) @updatedAt

  email            String  @unique
  stripeCustomerId String? @unique
}
```

If this is the issue, then I'm not sure whether it's possible to tell Prisma to store id as `uuid` type. Is there some way to convert `uuid` to `text` or vice versa in the RLS, to make the comparison?

**Comment by GaryAustin1** [comment]
You declared your id column to be type text.  Although you can force typing in your RLS with ::uuid, it would be better to change the column to type uuid.

**Reply by lumenwrites** [reply]
Thank you for your reply! Do you know how I can do that in `prisma.schema`? I still can't find a way to declare a `uuid` type.

**Reply by anttituomola** [reply]
You can use @db.uuid like this:

id        String   @id @default(dbgenerated("gen_random_uuid()")) @db.Uuid

**Reply by rizkysyahri** [reply]
hi, does this work? I currently have the same case, when the user logs in the user data doesn't enter my table editor, just like this.

**Comment by GaryAustin1** [comment]
I don't know much about prisma.  To keep moving you should be able to add id::uuid to your RLS for now.

**Comment by anttituomola** [comment]
I'm bit late here, but I had the same issue, and that got solved by putting the id inside of quotes, so

`USING auth.id() = "id"`

**Reply by chescript** [reply]
This worked for me 
Thank you

**Comment by abdifitahabdulkadir** [comment]
> Hi! I have a `Profile` table that contains all the data I want to store about the logged in user. Every time a user signs up to my website using Supabase Auth, Postgres trigger function creates a new row in the `Profile` table, with an `id` matching the `id` of the new user. Here's what profile table looks like: <img alt="Screen Shot 2022-07-15 at 02 32 37" width="1118" src="https://user-images.githubusercontent.com/4761235/179128882-b126f3ee-ab6c-484a-97cf-7541b8b94229.png">
> 
> Now my goal is to create a RLS policy that allows logged in user to fetch their profile data. Here's what I'm trying to do: <img alt="Screen Shot 2022-07-15 at 02 29 58" width="1131" src="https://user-images.githubusercontent.com/4761235/179129014-8b566d2d-dd2f-4d80-8151-21602765aca5.png"> <img alt="Screen Shot 2022-07-15 at 02 30 15" width="1145" src="https://user-images.githubusercontent.com/4761235/179129020-0e9a409c-1b48-454c-bbb4-eca7be88aa7b.png">
> 
> But when I'm trying to save this policy, I'm getting an error: <img alt="Screen Shot 2022-07-15 at 02 34 52" width="352" src="https://user-images.githubusercontent.com/4761235/179129101-f494498c-e53d-4418-9b05-b9f0bdeeb68b.png">
> 
> I'm new to RLS and I don't understand what am I doing wrong. Can you please help me to figure out how to do this correctly?
> 
> Maybe there's some kind of mismatch of types between `Profile.id` and `auth.users.id`? In that case, how can I make them match? The `Profile` table is created using Prisma, like so:
> 
> ```
> model Profile {
>   id        String   @id @default(cuid())
>   createdAt DateTime @default(now())
>   updatedAt DateTime @default(now()) @updatedAt
> 
>   email            String  @unique
>   stripeCustomerId String? @unique
> }
> ```
> 
> If this is the issue, then I'm not sure whether it's possible to tell Prisma to store id as `uuid` type. Is there some way to convert `uuid` to `text` or vice versa in the RLS, to make the comparison?

Hi If you are using supabase 2024 then I found this solution for me and worked thanks you 

![Screenshot 2024-07-22 182506](https://github.com/user-attachments/assets/0a57fc07-46cf-4fee-9321-abbffcd0d949)
